pkg/controller/serviceexternalip/controller.go

```diff
@@ -155,6 +155,12 @@ func (c *ServiceExternalIPController) enqueueService(obj interface{}) {
 		Namespace: service.Namespace,
 		Name:      service.Name,
 	}
+
+	_, allocationExists := c.getExternalIPAllocation(namespacedName)
+	if !allocationExists && getServiceExternalIPPool(service) == "" {
+		return
+	}
```

Comment:
could there be a race condition with a concurrent call to `syncService` operating on a previous version of the Service?
1) `enqueueService` (this function) observes no current allocation (`allocationExists` is false) and no annotation (`getServiceExternalIPPool(service)` is the empty string)
2) `syncService` observes a previous version of the Service with the annotation, and is in the process of creating an allocation.

@tnqn do you know if it is safe to have a stateful check in this code path (event handler function)?

Reply:
updated the checks for annotations to `ResourceEventHandlerFuncs` before `enqueueService` as it is not needed to check the IP allocations.

Reply:
@antoninbas I discussed with @xliuxu offline, there is no need to have a stateful check in event handlers.